### PR TITLE
Log Unsupported TF Variables

### PR DESCRIPTION
**Pull request by NihalHarish**
### Description of changes:
- Add a string output to the NotImplemented Error so that it helps us identify the exact type of variable that causes failures.

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smdebug/tensorflow/tensor_ref.py [hunk 1]
```diff
@@ -22,7 +22,9 @@ def get_tf_names(arg):
     elif isinstance(arg, values.DistributedVariable):
         tf_names = [v.name for v in arg._values]
     else:
-        raise NotImplementedError
+        raise NotImplementedError(
```
Comment (Vikas-kum) [inline smdebug/tensorflow/tensor_ref.py:25]: should we just support distributed values I don't see things getting worse than what it is today.
a  proposal is: 
else:
  #log
  if instance of DistributedValue : 
      tf_names = [v.name for v in arg._values] 
 else:
   Not implemented error

Comment (NihalHarish) [reply smdebug/tensorflow/tensor_ref.py:25]: Done.

File: smdebug/tensorflow/tensor_ref.py [hunk 2]
```diff
@@ -19,10 +19,12 @@ def get_tf_names(arg):
         tf_names = [arg.name]
     elif isinstance(arg, tf.Tensor):
         tf_names = [arg.name]
-    elif isinstance(arg, values.DistributedVariable):
+    elif isinstance(arg, values.DistributedValues):
```
Comment (vandanavk) [inline smdebug/tensorflow/tensor_ref.py:22]: make sure you replace all occurrences of DistributedVariable. also, make sure it doesn't break the training script reported on a recent internal ticket

Comment (NihalHarish) [reply smdebug/tensorflow/tensor_ref.py:22]: Done. Thanks for this

**Comment by NihalHarish** [comment]
> we should probably have a test or a sample training script for this case.

Don't the current tests and examples execute this same code path.

**Comment by vandanavk** [comment]
> > we should probably have a test or a sample training script for this case.
> 
> Don't the current tests and examples execute this same code path?

Doesn't look like the current tests test all child classes of DistributedValues.

**Comment by Vikas-kum** [comment]
> > > we should probably have a test or a sample training script for this case.
> > 
> > 
> > Don't the current tests and examples execute this same code path?
> 
> Doesn't look like the current tests test all child classes of DistributedValues.

Yeah, seee if there i

**Pull request merged**